pkg/webhook/sidecar_spec.go

```
@@ -204,9 +204,18 @@ func (si *SidecarInjector) GetMetadataPrefetchSidecarContainerSpec(pod *corev1.P
 			enableMetaPrefetchRaw, ok := volumeAttributes[gcsFuseMetadataPrefetchOnMountVolumeAttribute]
 			// We disable metadata prefetch by default, so we skip injection of volume mount when not set.
 			if !ok {
-				continue
+				// volume did not enable metadata prefetch, checking if sc enables metadata prefetch
+				sc, err := si.GetVolumesStorageClass(pv)
+				if err != nil {
```

We need to distinguish on whether the StorageClass is `""` (this is clearly a user error if they are using gcsfuse profiles) and whether the SC was not available (this could happen for a variety of reasons, most likely a race condition where the Pod spec is created before the StorageClass).

The user error should return an error and inform the customer.

The second thing I mentioned is best effort, and is not handled rigurously in the webhook anyway, so you can just keep this warning.